insert-markdown.awk

```diff
@@ -1,12 +1,16 @@
 # n -- is to work on the nth code block
 # rel_path -- is the relative path to the file
-# hide_codeblocks -- is a flag to hide the markup with a details tag
+# codeblock_action -- what to do with the codeblock
 
 BEGIN {
   i=0;
   start_comment="<!-- generated by mermaid compile action - START -->";
   end_comment="<!-- generated by mermaid compile action - END -->";
   link="![~mermaid diagram " n "~](" path ")";
+  # For inlining you need HTTP headers.
+  # These seem only to get provided by e.g. raw.github.com.
+  # See https://stackoverflow.com/questions/13808020/include-an-svg-hosted-on-github-in-markdown
+  inline="<img src=\"https://raw.githubusercontent.com/" GITHUB_REPOSITORY "/" GITHUB_SHA  path "\">";
```

Comment:
This seems reasonable, but I don't think we can get an SHA for a commit before the commit is made. See related comment.

Reply:
I think the best way to solve this is to split `main` into stages:
1. build all refs
2. commit refs
3. modify Markdown with committed sha

Reply:
That does seem what has to happen in order to reference an SHA for the image URL, what I'd like to do then is rather than target these PRs at `master` merge them on a branch until we the feature is done.

Reply:
I've created the V2 branch for this, please close this pr, open it to the branch, then can continue work there.

Reply:
Just changed to base of `V2` and can continue discussing here.

Reply:
Sounds good

Reply:
Soo surprise. I just today tried a manual relative `img` reference to `png`s today and it works fine. No crazy stages necessary.
Thus I reinstate this PR to `master` :)

Reply:
Okay, so this looks like you dropped the SHA reference from the image src path, ya? I'll test this out tonight, then likely merge